torchinductor/codegen/triton.py

```diff
@@ -274,17 +281,44 @@ def __init__(
     def codegen_next(self):
         return self.name
 
-    def simplify(self, expr, nodes):
-        return expr
-
-    def construct(self, lengths):
+    def construct(self, lengths: List[sympy.Expr]):
         node = self
         itervars = []
         for sv in reversed(lengths):
             node = node.child_node(sv)
             itervars.append(node.symbol())
+        self.compute_recursive_names()
         return list(reversed(itervars))
 
+    def vars_and_sizes(self, index: sympy.Expr):
+        """Figure out vars from this tree used in index"""
+        symbol_names = set(map(str, index.free_symbols)) & self.recursive_names
```

Comment:
can you sort this set so that kernels still have deterministic source?

Reply:
This one is only used for `set.union` calls, so I think we want to keep it a set.  The order comes from the tree structure, not from this variable.  I'll change the other one though.